joplin/base/signals/janis_build_triggers.py

```diff
@@ -1,131 +1,212 @@
+import requests, json
+from datetime import datetime
+from pytz import timezone
 from django.conf import settings
 from django.dispatch import receiver
 from django.db.models.signals import post_save, post_delete
 from wagtail.core.signals import page_published, page_unpublished
 from graphene import Node
 from wagtail.core.models import Page
 from wagtail.admin.models import get_object_usage
+from rest_framework_api_key.models import APIKey
 
 from snippets.contact.models import Contact
 from pages.guide_page.models import GuidePage
-from groups.models import Department
 from wagtail.documents.models import Document
-from base.signals.aws_publish import get_http_request, create_build_aws
-from base.signals.netlify_publish import netlify_publish
-from base.signals.publish_v3 import publish_v3
-from flags.state import flag_enabled
+from pages.home_page.models import HomePage
+
 
 import logging
 logger = logging.getLogger('joplin')
 
-JANIS_SLUG_URL = settings.JANIS_SLUG_URL
-
-
-def trigger_build(sender, pages_ids, action='saved', instance=None):
-    """
-    triggers different build process depending on environment
-    source = name of snippet or object triggering build
-    """
-    trigger_object = instance
-    logger.info(f'{trigger_object} {action}, triggering build')
-    if settings.IS_STAGING or settings.IS_PRODUCTION or settings.IS_REVIEW:
-        publish_v3(pages_ids)
 
-def collect_pages(instance):
+'''
+Gets data from a page that is required to send to the Publisher.
+
+returns {
+    id: Int, the page_id used by wagtail.
+    global_id: String, a hashed id used to query against graphql api
+    triggered_build: Boolean, was this the page that triggered the publish request?
+    action:
+        "published" - publish action for itself and other pages
+        "unpublished" - this page triggered the unpublish action for itself and other pages
+        "updated_by_snippet" - this page was updated by a snippet being saved or deleted
+        "saved" - a snippet is being saved, has no impact on Janis itself, but it could result in pages getting "secondary_publish_by_snippet"
+        "deleted" - a snippet is being deleted, has no impact on Janis itself, but it could result in pages getting "secondary_publish_by_snippet"
+    is_page: Boolean, is it a page as opposed to a snippet?
+    content_type: String, which specific content_type (this is more for logging)
+    author: Int, the id of the author of the latest revision
+}
+TODO: Figure out what format global_id should be in order to run queries with it.
```

Comment:
i think these are less complex in grapple

Reply:
Ah good to know